velero-plugins/migpod/stage.go

```diff
@@ -0,0 +1,71 @@
+package migpod
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/fusor/openshift-migration-plugin/velero-plugins/migcommon"
+	"github.com/sirupsen/logrus"
+	corev1API "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func buildStagePod(pod *corev1API.Pod, log logrus.FieldLogger) *corev1API.Pod {
+	log.Infof("[pod-restore] creating stage pod for %s", pod.Name)
+
+	// Map of Restic volumes.
+	resticVolumes := map[string]bool{}
+	for _, name := range strings.Split(pod.Annotations[migcommon.MigrateQuiesceAnnotation], ",") {
+		resticVolumes[name] = true
+	}
+	// Base pod.
+	newPod := corev1API.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       pod.Kind,
+			APIVersion: pod.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: pod.Namespace,
+			Name:      pod.Name + "-" + "stage",
+			Annotations: map[string]string{
+				migcommon.MigrateQuiesceAnnotation: pod.Annotations[migcommon.MigrateQuiesceAnnotation],
+			},
+			Labels: pod.Labels,
+		},
+		Spec: corev1API.PodSpec{
+			Containers:                   []corev1API.Container{},
+			Volumes:                      []corev1API.Volume{},
+			SecurityContext:              pod.Spec.SecurityContext,
+			ServiceAccountName:           pod.Spec.ServiceAccountName,
+			AutomountServiceAccountToken: pod.Spec.AutomountServiceAccountToken,
+		},
+	}
+
+	// Set NodeName
+	newPod.Spec.NodeName = pod.Spec.NodeName
```

Comment:
We don't want to set this here. With stage pods on src, this was to ensure that the stage pod was on the same node as the application pod, since some volume plugins required that. In this case, we don't want NodeName set, since setting a src cluster NodeName will result in a pod that can't be scheduled on dest, since this node won't exist.